app/Http/Requests/Contracts/HasOptionalUser.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Http\Requests\Contracts;
+
+use App\Models\User;
+
+interface HasOptionalUser
+{
+	public const USER_ID_ATTRIBUTE = 'userID';
+
+	/**
+	 * Returns an _additional_ {@link User} object associated with this request.
+	 *
+	 * This method is called `user2`, because Laravel already defines
+	 * {@link \Illuminate\Http\Request::user()} which returns the user which
+	 * is currently authenticated within the HTTP session.
+	 * This method returns another user object which is explicitly part of the
+	 * request.
+	 *
+	 * @return User|null
+	 */
+	public function user2(): ?User;
```

Comment:
rather than `optionalUser` or `requestUser` I would go for `modifiedUser`.

- _optional_ does not make sense on the `HasUser` interface which extends this one.
- request still induces ambiguity: is it the user _doing_ the request or the user _modified by_ the request?

Reply:
The term `user2` hasn't been introduced by this PR. It has been there before. Of course, this is not an argument to not try to improve it, but I don't have any good idea. @ildyria already gave the same thoughts I had.

We could call it `userToBeModified` (very clumsy, I don't like it, but unambitious). `additionalUser` but I don't think this is better than `user2`.

Reply:
@kamil4 If your are fine with this, you should resolve the conversation.